picard/ui/profileeditor.py

```diff
@@ -0,0 +1,455 @@
+        self.ui.buttonbox.addButton(self.new_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
+
+        self.copy_profile_button = QtWidgets.QPushButton(_('Copy'))
+        self.copy_profile_button.setToolTip(_("Copy to a new profile"))
+        self.copy_profile_button.clicked.connect(self.copy_profile)
+        self.ui.buttonbox.addButton(self.copy_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
+
+        self.delete_profile_button = QtWidgets.QPushButton(_('Delete'))
+        self.delete_profile_button.setToolTip(_("Delete the profile"))
+        self.delete_profile_button.clicked.connect(self.delete_profile)
+        self.ui.buttonbox.addButton(self.delete_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
+
+        self.save_profile_button = QtWidgets.QPushButton(_('Save'))
+        self.save_profile_button.setToolTip(_("Save changes to the profile"))
+        self.save_profile_button.clicked.connect(self.save_profile)
+        self.ui.buttonbox.addButton(self.save_profile_button, QtWidgets.QDialogButtonBox.ActionRole)
+
+        self.cancel_button = QtWidgets.QPushButton(_('Cancel'))
+        self.cancel_button.setToolTip(_("Close the profile editor without saving changes to the profiles"))
+        self.ui.buttonbox.addButton(self.cancel_button, QtWidgets.QDialogButtonBox.RejectRole)
+        self.ui.buttonbox.rejected.connect(self.close)
+
+        self.ui.buttonbox.addButton(QtWidgets.QDialogButtonBox.Help)
+        self.ui.buttonbox.helpRequested.connect(self.show_help)
+
+    def load(self):
+        """Load initial configuration.
+        """
+        config = get_config()
+        self.profile_settings = config.profiles[self.SETTINGS_KEY]
+
+        for profile in config.profiles[self.PROFILES_KEY]:
+            list_item = ProfileListWidgetItem(profile['title'], profile['enabled'], profile['id'])
+            self.ui.profile_list.addItem(list_item)
+
+        # Select the last selected profile item
+        last_selected_profile_pos = config.persist[self.POSITION_KEY]
+        self.expanded_sections = config.persist[self.EXPANDED_KEY]
+        last_selected_profile = self.ui.profile_list.item(last_selected_profile_pos)
+        settings = None
+        if last_selected_profile:
+            self.ui.profile_list.setCurrentItem(last_selected_profile)
+            last_selected_profile.setSelected(True)
+            id = last_selected_profile.profile_id
+            self.current_profile_id = id
+            settings = self.get_settings_for_profile(id)
+        self.make_setting_tree(settings=settings)
+
+    def get_settings_for_profile(self, id):
+        """Get the settings for the specified profile ID.
+
+        Args:
+            id (str): ID of the profile
+
+        Returns:
+            dict: Profile settings
+        """
+        # Add empty settings dictionary if no dictionary found for the profile.
+        if id not in self.profile_settings:
+            self.profile_settings[id] = {}
```

Comment:
I'm not fond of a `get` method modifying things.
The comment tells what it does, but doesn't explain why it is needed. Can you clarify?

Reply:
It was actually to avoid a KeyError exception if that element was missing.  This happens when a new profile is created.  I'll change it to:

```
return self.profile_settings[id] if id in self.profile_settings else {}
```

and add some code to make sure that the key is created when a new element is created.  To do that I'll return the ID from `ProfileListWidget.add_profile()`.  That actually makes more sense than the way I'm currently doing it.

Reply:
Actually, it's a lot more of an issue to change than I expected because other code is triggered when the `ProfileListWidget.add_profile()` is executed which also relies on this key having been created.  I'll add more of an explanation of what's happening to the `get_settings_for_profile()` method.